src/Tribe/Tickets.php

```diff
@@ -176,6 +176,113 @@ public function get_ticket( $event_id, $ticket_id ) {
 		}
 
 		/**
+		 * Returns the html for the delete ticket link
+		 *
+		 * @since TBD
+		 *
+		 * @param obj $ticket Ticket object
+		 */
+		public function get_ticket_delete_link( $ticket = null ) {
+			if ( empty( $ticket ) ) {
+				return;
+			}
+
+			if ( apply_filters( 'tribe_tickets_current_user_can_delete_ticket', true, $ticket->ID, $ticket->provider_class ) ) {
+				$delete_link = sprintf(
+					'<span><a href="#" attr-provider="%1$s" attr-ticket-id="%2$s" id="ticket_delete_%2$s" class="ticket_delete">'
+					. esc_html__( 'Delete Ticket', 'event-tickets' ) . '</a></span>',
+					$ticket->provider_class,
+					$ticket->ID
+				);
+
+				return $delete_link;
+			}
+
+			return;
+		}
+
+		/**
+		 * Returns the url for the move ticket link
+		 *
+		 * @since TBD
+		 *
+		 * @param int $post_id the id of the parent post/event
+		 * @param obj $ticket Ticket object
```

Comment:
Throughout this change we should switch from undefined type `obj` to the generic `object type